xcube_stac/stac.py

```diff
@@ -0,0 +1,65 @@
+# The MIT License (MIT)
+# Copyright (c) 2024 by the xcube development team and contributors
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import xarray as xr
+
+
+class Stac:
+    """ Class containing methods handling STAC catalogs
+    """
+
+    def __init__(
+        self, url: str,
+        collection_prefix: str = None,
+        data_id_delimiter: str = "/"
+    ):
+        """
+        Args:
+            url (str): URL to STAC catalog
+            collection_prefix (str, optional): Path of collection used as
+                entry point. Defaults to None.
+            data_id_delimiter (str, optional): Delimiter used to separate
+                collections, items and assets from each other.
+                Defaults to "/".
+        """
```

Comment:
In most xcube classes, `__init__` does not have a docstring. Instead, argument documentation is part of the class' docstring, because in 99.99% of all cases `__init__` is called to create instances of the class.